exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,376 @@
+The parser must recognize those white spaces as separator(s).
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Leading White Spaces Options Parsing
+`req~leading-white-spaces-script-options-parsing~1`
+
+The parser must accept Script Options for lines starting with any number of white space characters before the Script Options.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Ignore anything which is not a Script Option
+`req~ignore-none-script-options~1`
+
+If there is any character in front of a Script Option which is not a white space, the parser must ignore the option(s). 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Line Script Options Parsing
+`req~multiple-lines-script-options-parsing~1`
+
+The parser must recognize Script Options at any line in the given script code.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### White Spaces Options Parsing V1
+`req~white-spaces-script-options-parsing-v1~1`
+
+All white spaces between the option key and option value are to be interpreted as separator. 
+White spaces between the option value and the terminating ";" are to be removed from the option value.
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### White Spaces Options Parsing V2
+`req~white-spaces-script-options-parsing-v2~1`
+
+All white spaces between the option key and option value are to be ignored. The following rules for escape sequences at **the start** of a script optionValue are to be applied:
+- '\ ' => space character
+- '\t' => <tab> character
+- '\f' => <form feed> character
+- '\v' => <vertical tab> character
+
+White spaces between the option value and the terminating ";" shall be interpreted as part of the value.
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Options
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key.
+
+Comment: 
+"Collecting" in this context means that a merging strategy must be applied. We don't want multiple options with the same key to result in simply overwriting or discarding values. Please note that the specific handling depends on the individual option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Duplicate Options Management
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key and same/different value. Note: the specific handling depends on the option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Script Option Removal
+`req~script-option-removal~1`
+
+The parser handler must remove found known Script Options from the original script code. 
```

Comment:
Hm, this implemented by the handler, do we have a reusable implementation for that, otherwise we need to reimplement this for each handler

Reply:
Not yet, but should be an easy refactoring to do so.